tests/unit/test_spatial.py

```diff
@@ -182,6 +183,126 @@ def test_get_spatial_index_ranges_geographic(self):
+            }
+        )
+
+        x_y_extents = {
+            'x_min': 192972.56050179302,
+            'x_max': 4052423.7705376535,
+            'y_min': 5930779.396449475,
+            'y_max': 6979878.9118312765,
+        }
+
+        mock_get_x_y_extents.return_value = x_y_extents
+
+        # When ranges are derived, they are first calculated for x, then y:
+        mock_get_dimension_index_range.side_effect = [(487, 595), (9, 38)]
+
+        with self.subTest(
+            'Projected grid from coordinates gets expected dimension ranges'
+        ):
+            with Dataset(smap_file_path, 'r') as smap_prefetch:
+                self.assertDictEqual(
+                    get_x_y_index_ranges_from_coordinates(
+                        '/Soil_Moisture_Retrieval_Data_AM/surface_flag',
+                        smap_varinfo,
+                        smap_prefetch,
+                        latitude_coordinate,
+                        longitude_coordinate,
+                        {},
+                        bounding_box=bbox,
+                        shape_file_path=None,
+                    ),
+                    expected_index_ranges,
+                )
+
+                mock_get_x_y_extents.assert_called_once_with(
+                    ANY, ANY, crs, shape_file=None, bounding_box=bbox
+                )
+
+                self.assertEqual(mock_get_dimension_index_range.call_count, 2)
+                mock_get_dimension_index_range.assert_has_calls(
+                    [
+                        call(
+                            ANY,
+                            x_y_extents['x_min'],
+                            x_y_extents['x_max'],
+                            bounds_values=None,
+                        ),
+                        call(
+                            ANY,
+                            x_y_extents['y_min'],
+                            x_y_extents['y_max'],
+                            bounds_values=None,
+                        ),
+                    ]
+                )
+
+        mock_get_x_y_extents.reset_mock()
+        mock_get_dimension_index_range.reset_mock()
+
+        with self.subTest('Function does not rederive known index ranges'):
+            with Dataset(smap_file_path, 'r') as smap_prefetch:
+                self.assertDictEqual(
+                    get_x_y_index_ranges_from_coordinates(
+                        '/Soil_Moisture_Retrieval_Data_AM/surface_flag',
+                        smap_varinfo,
+                        smap_prefetch,
+                        latitude_coordinate,
+                        longitude_coordinate,
+                        expected_index_ranges,
+                        bounding_box=bbox,
+                        shape_file_path=None,
+                    ),
+                    {},
+                )
+
+            mock_get_x_y_extents.assert_not_called()
+            mock_get_dimension_index_range.assert_not_called()
+
```

Comment:
This seems a bit overly complicated for what is a relatively simply stated function.  I would have imagined a predefined CRS, and a list of lat/lon points, plus a corresponding set of expected x, y results.  Perhaps two calls with the second call having an additional data-point / result-point, and a check on mock-subs call counts to ensure caching is properly implemented.